Tasks/VsTest/settingshelper.ts

```diff
@@ -55,89 +52,89 @@ const runSettingsTemplate = `<?xml version=\"1.0\" encoding=\"utf-8\"?>
     </DataCollectionRunSettings>
     </RunSettings>`;
 
-export async function updateSettingsFileAsRequired(settingsFile: string, isParallelRun: boolean, tiaConfig: models.TiaConfiguration, vsVersion: version.VSTestVersion, videoCollector: boolean, overrideParametersString: string, isDistributedRun: boolean): Promise<string> {
-    const defer = Q.defer<string>();
-    let result: any;
 
-    if (!isParallelRun && !videoCollector && !tiaConfig.tiaEnabled && !overrideParametersString) {
-        defer.resolve(settingsFile);
-        return defer.promise;
-    }
+export function updateSettingsFileAsRequired(settingsFile: string,
+    isParallelRun: boolean,
+    tiaConfig: models.TiaConfiguration,
+    vsVersion: version.VSTestVersion,
+    videoCollector: boolean,
+    overrideParametersString: string,
+    isDistributedRun: boolean): string {
+    try {
+        let result: any;
+        let parser = new xmlParser.Parser();
 
-    //Get extension of settings file and contents
-    let settingsExt = null;
-    if (settingsFile && fs.lstatSync(settingsFile).isFile() && settingsFile.split('.').pop().toLowerCase() === 'testsettings') {
-        settingsExt = testSettingsExtension;
-        result = await utils.Helper.getXmlContents(settingsFile);
-        if (!result || result.TestSettings === undefined) {
-            tl.warning(tl.loc('InvalidSettingsFile', settingsFile));
-            settingsExt = null;
+        if (!isParallelRun && !videoCollector && !tiaConfig.tiaEnabled && !overrideParametersString) {
+            return settingsFile;
         }
-    } else if (settingsFile && utils.Helper.pathExistsAsFile(settingsFile)) {
-        settingsExt = runSettingsExtension;
-        result = await utils.Helper.getXmlContents(settingsFile);
-        if (!result || result.RunSettings === undefined) {
+
+        let settingsExt = null;
+        if (settingsFile && fs.lstatSync(settingsFile).isFile() && settingsFile.split('.').pop().toLowerCase() === 'testsettings') {
+            settingsExt = testSettingsExtension;
+        }
+        else if (settingsFile && utils.Helper.pathExistsAsFile(settingsFile)) {
+            settingsExt = runSettingsExtension;
+        }
+
+        if (settingsFile) {
+            result = utils.Helper.getXmlContentsSync(settingsFile);
+        }
+
+        if (!result || (result.TestSettings === undefined && result.RunSettings === undefined)) {
```

Comment:
> i [](start = 7, length = 2)

If a runsettings xml is named as .testsettings, or vice versa, will this catch that as invalid

Reply:
No this will not. Adding a check in this case
`if (result && (settingsExt == runSettingsExtension && result.RunSettings === undefined) || 
                      (settingsExt == testSettingsExtension && result.TestSettings === undefined)) {
            throw new Error(tl.loc('InvalidSettingsExtension', settingsFile));
        }`
This should handle it.